### ISSUE TITLE
sl add remote does not work

### ISSUE THREAD

**Issue by kibagare** [facebook/sapling#777]
I have a local sl repository which I want to push to github.

The [basic commands](https://sapling-scm.com/docs/overview/basic-commands/) have nothing on `push`.
The [push documentation](https://sapling-scm.com/docs/commands/push/) seems to assume that you have a remote added already.

The [Git cheat sheet](https://sapling-scm.com/docs/introduction/git-cheat-sheet/#working-with-commits) says that the equivalent of `git remote add REMOTE URL` is `sl path --add REMOTE URL`.
When I do that, I get `abort: invalid URL - invoke as 'sl paths -a NAME URL`'.
When I do `sl paths -a <my url>` I get the same error message.

How does this work?

**Comment by sggutier** [comment]
Can you share what you are passing as `REMOTE` and as `URL`? Usually I run it with something like

```
sl path --add personalfork https://github.com/sggutier/sapling
```

**Comment by kibagare** [comment]
after initialising sl and making a commit, made a new repository on github, then:
```
sl path --add main https://github.com/kibagare/distances.git
sl push --to main
```

which gives:
`abort: use '--to' to specify destination bookmark`

**Comment by muirdm** [comment]
The `[DEST]` is a positional argument, so `sl push main --to <branch name>`.

`sl push --help -v` has one illustrative example "hg push --rev 05a82320d my-fork --to my-branch --force", but that should not be so buried. We also should rethink the "--to" option since it is so ambiguous.

**Comment by kibagare** [comment]
Adding a remote and pushing to it should be as simple as possible. This sounds more complicated than git, which defies the whole idea of developing a simple version control system. I hope we can come up with something more compact than this.

**Comment by karimelazzouni** [comment]
Agreed that this is very confusing. Just spent about an hour trying to push to an existing github repository. I had to use `git`'s workflow at the end. Also, when I finally got my repo in order and wanted to pull the commits I just did with `sl`, I couldn't pull the whole history (only the last commit) because arguments to `sl pull` like `--unshallow` don't exist.
